pkg/handler/eventhandler_suite_test.go

```diff
@@ -34,9 +35,10 @@ func TestEventhandler(t *testing.T) {
 
 var testenv *envtest.Environment
 var cfg *rest.Config
+var logBuffer bytes.Buffer
```

Comment:
Could this global variable cause issues since it's mutated (`Reset`) in the test specs? How about unit testing the logging in `enqueueRequestForOwner` instead of in the e2e-tests?

Reply:
> How about unit testing the logging in enqueueRequestForOwner instead of in the e2e-tests?

The logger is currently getting tested while unit testing for specific events in `enqueueRequestForOwner`. Could you point to the alternate option?

> Could this global variable cause issues since it's mutated (Reset) in the test specs?

Would prefer to keep this at the beginning of test suite, because it needs to be set as the [default logger](https://github.com/varshaprasad96/controller-runtime/blob/0edc5c2afb51816e2834bec25dc4838ef5fe67c5/pkg/handler/eventhandler_suite_test.go#L41) before starting the tests. `BeforeSuite` anyways gets executed only once, though we are resetting the content quite a few times in the test spec for specific tests.